distributednetworkmanager/src/main/java/com/gruppo1/distributednetworkmanager/DistributedNetworkActionResponder.java

```diff
@@ -0,0 +1,90 @@
+package com.gruppo1.distributednetworkmanager;
+
+import android.content.Context;
+import android.telephony.SmsMessage;
+
+import com.dezen.riccardo.smshandler.SMSHandler;
+import com.dezen.riccardo.smshandler.SMSManager;
+import com.dezen.riccardo.smshandler.SMSMessage;
+
+public class DistributedNetworkActionResponder {
+
+    private Context context;
+
+
+    public DistributedNetworkActionResponder(Context context){
+
+        this.context = context;
+    }
+
+    /**
+     * This method is called when a message with an action is received
+     *
+     * @param action received action
+     */
+    public void onActionReceived(KadAction action) {
+        switch (action.getActionType()) {
```

Comment:
Here, since you only care about Requests, can add an if to return immediately when you receive a Response (there's a method in the ActionType enum called .isResponse() for that). Remember: you get Requests, you send Responses, that's this class' job